internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,29 +116,177 @@ func TestDelayConsumer_Run(t *testing.T) {
 	dc.Run(ctx)
 }
 
-func getDummyBrokerMessage() messagebroker.ReceivedMessage {
-	tNow := time.Now()
-	tPast := tNow.Add(time.Second * 100 * -1)
+func TestDelayConsumer_Run_Consume(t *testing.T) {
+	// message with delay within duration of test-case
+	msg1 := getDummyBrokerMessage("msg-1", uuid.New().String(), time.Now().Add(time.Nanosecond*1000), 5)
+	// message with delay outside duration of test-case
+	msg2 := getDummyBrokerMessage("msg-2", uuid.New().String(), time.Now().Add(time.Second*2), 5)
+
+	type Test struct {
+		name                 string
+		messages             *messagebroker.ReceivedMessage
+		expectedMessage      *messagebroker.ReceivedMessage
+		expectedMessageCount int
+	}
+
+	tests := []Test{
+		{
+			name:                 "DelayConsumer with message delivery time within test duration",
+			messages:             &msg1,
+			expectedMessage:      nil,
+			expectedMessageCount: 0,
+		},
+		{
+			name:                 "DelayConsumer with message delivery time above test duration",
+			messages:             &msg2,
+			expectedMessage:      &msg2,
+			expectedMessageCount: 1,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// since new context is needed to for each time so set up is called each time
+			ctx, ctrl, subCtx, cancel, brokerStore, handler, consumer, cache := setupDC(t)
+			subs := getMockSubscriptionModel()
+			subscriberID := "subscriber-id"
+			errCh := make(chan error)
+			dc, err := NewDelayConsumer(subCtx, subscriberID, "t1", subs, brokerStore, handler, cache, errCh)
+			assert.NotNil(t, dc)
+			assert.Nil(t, err)
+			assert.NotNil(t, dc.LogFields())
+
+			producer := getMockProducer(ctx, ctrl)
+			brokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
```

Comment:
why didn't we move this to setupDC..?

Reply:
@sakshi-goyal-razorpay why these needs to be setupDC? if we will put everything in setup and their parameters in setupDC parameters will keep on increasing going further and we will need refactor the code again again. we should expect some of the calls in test case according to use case